tests/ui/parser/bad-name.stderr

```diff
@@ -9,6 +9,12 @@ error: expected a pattern, found an expression
    |
 LL |   let x.y::<isize>.z foo;
    |       ^^^^^^^^^^^^^^ arbitrary expressions are not allowed in patterns
+   |
+help: remove the `let` if you meant to evaluate an expression
```

Comment:
I wonder if `let <expr>;` ever comes up in practice and if the suggestion is actually what the user actually intended. I guess it's fine